### PR TITLE
add testing environment

### PR DESCRIPTION
**Pull request by sgratzl**
**Prerequisites**:

- [x] Unless it is a hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary

 * adds https://jestjs.io/ for unit-testing together with https://testing-library.com/docs/svelte-testing-library/intro -> see .spec files in src directory
 * adds https://www.cypress.io/ for e2e testing, see cypress/integrations spec files for examples
 * small refactoring for easier testing 

usage -> see README section

### CODE REVIEW

**Comment by dlaliberte** [comment]
I reviewed the jest and cypress web sites and docs, and it all looks good to me.  The main question I would have is whether they work well with Svelte, given it's particular design characteristics.

**Comment by sgratzl** [comment]
for cypress it doesn't make a difference since it is working with the website. for Jest, I use https://testing-library.com/docs/svelte-testing-library/intro which should deal with Svelte characteristics

**Review by tildechris — COMMENTED** [review]
The testing framework here LGTM, but there's a lot to review.  

I think I understand the test hooks and writing tests, but I do not understand the changes around `msa.csv.js` and friends.  Can you summarize?

**Comment by sgratzl** [comment]
> I think I understand the test hooks and writing tests, but I do not understand the changes around `msa.csv.js` and friends. Can you summarize?

importing non JavaScript files (like CSV) in JavaScript files usually requires a bundler like webpack or rollup with a custom loader. However, now we also have Jest in the loop which compiles the test code and thus also needs to handle to those kind of imports files. To avoid a more complicated test setup, I wrapped the CSV files to JavaScript modules, similar to what the current raw-loader of webpack was doing before.

**Review by tildechris — APPROVED** [review]
That sounds reasonable.  LGTM, thanks for the work here.
